docs/cmd/kn_service_create.md

```diff
@@ -36,15 +36,20 @@ kn service create NAME --image IMAGE [flags]
 ### Options
 
 ```
+      --async                    Don't wait for the service to become Ready.
   -e, --env stringArray          Environment variable to set. NAME=value; you may provide this flag any number of times to set multiple environment variables.
       --force                    Create service forcefully, replaces existing service if any.
   -h, --help                     help for create
       --image string             Image to run.
       --limits-cpu string        The limits on the requested CPU (e.g., 1000m).
       --limits-memory string     The limits on the requested CPU (e.g., 1024Mi).
+      --max-scale int            Maximal number of replicas.
+      --min-scale int            Minimal number of replicas.
   -n, --namespace string         List the requested object(s) in given namespace.
       --requests-cpu string      The requested CPU (e.g., 250m).
       --requests-memory string   The requested CPU (e.g., 64Mi).
+      --target int               Number of in-flight requests per pod.
```

Comment:
The issues talks about containerConcurrency, so I think this should be named something like that. The "target" config option is different - we can add a flag for that too if people want it, but my original request is for CC.

Reply:
tbh, I think we can add all autoscaling processing annotations in one go. But I'm fine of course to remove `target` if we agree that this is not something we want to expose via the CLI (althouhg I believe every part of the public Knative API which is put into annotation *should* be available as CLI option as this is hard to find out elsewhere as it is not part of the CRD schema and nobody memorizes the full name of annotation keys)

Reply:
I'm +1 for exposing all of them too - I just didn't want this PR to go thru w/o addressing what the issue was asking for :-)

Reply:
got you, valid point.

Reply:
The name `target` (unless I am missing something or has historical meaning) is vague and not meaningful? Why can't this be `no-in-flights` or similar?

Reply:
Agreed, its a bit vague and just copier over literally from the annotation (`autoscaling.knative.dev/target`).  "no-in-flights" is probably also not ideal (I immediately, ah that's the opposite of `allow-in-flights` ;-). Let's find a better name ...

Reply:
I would suggest that we be consistent (even if not a 100% direct match) with what people will see in the yaml.  So, for now I would stick with `target` but suggest a new name for the yaml & serving code.

Having said that.... do I think `concurrency` might be good enough for the flag I was asking for - instead of `containerConcurrency`.  I think the word `container` is kind of redundant. Maybe serving can change this too at the same time  :-)

Reply:
were you going to add that flag?

Reply:
> were you going to add that flag?

These concurrency flags are added to `kn service create` and `kn service update`. There is the general problem that for changing these annotations on update, this will trigger a new revision, and don't touch the old revision. This means if you have used `minScale` then this revision will always spin up that many pods even when its not the latest revision anymore (revisions are immutable). The only thing to get this back to 0 is to fiddle around with the autoscaler CRD. There is some discussions over there to make metadata like annotations mutable on revision instances, but as long as this doesn't happen we have to deal with that (i.e. don't modify the annotation on the running Revision but on the Service/Configuration's Revision template). See https://github.com/knative/serving/issues/3547#issuecomment-498206979 for the discussion over there in serving.

Reply:
For the naming: 

I like `--min-scale` and `--max-scale` as it's concise and pretty clear what it means. Alternatively we could call it `--min-replicas` & `--max-replicas` to reflect the Deployment fields affected (or even `--min-pods` and `--max-pods`). However this suggestion again connects to K8s which we shouldn't do ("replicas" and even more "pods" is a concept occupied by K8s). So I'm fine with `--min-scale` & `--max-scale`.

Another option would be `--scale-max` and `--scale-min` which would put the options nicely together, e.g. in an alphabetically ordered help options list or documentation. This is important when your option list gets long (ever tried `jx create cluster minikube --help` ;-P), and I think for create and update it will still grow.

For `--target` I agree that it is to unspecific as an option name. 

My suggestions would be (with preference in that order): 

* `--scale-target 10` (would fit to a `--scale-min` and `--scale-max` (but could be misread that it's a target to reach 10, but afaiu its just the number of concurrent requests after that a scale up event happen)
* `--scale-after-requests 10` (which reads nicely like in `scale after 10 request`) 
* `--scale-after 10`
* `--concurrency-target 10`
* `--scale-in-flight 10`(not sure whether this clear enough)
* `--scale-after-concurrent-requests 10`

Reply:
For clarity, `containerConcurrency` and `target` are two different things. The first is an enforced limit, the second is a "soft" target for the autoscaler. In the absence of the second, the second is inferred from the first.

In that sense I'd opt for:

- `--concurrency-limit` to define the former
- `--concurrency-target` to define the latter. We should encode `concurrency` into here as for the KPA the default will be `concurrency`. If we also want to expose the HPA as an autoscaler, things will get more tricky here though.

I think that nicely encodes the difference in the name. The CLI docs could say that `--concurrency-target` defaults to `--concurrency-limit` if not passed.

Reply:
Ah, I see. There can be different types of autoscaling triggers, which currently is based on concurrent requests (but could be e.g. CPU consumption when using the HPA). But maxScale and minScale are more a global scaling (boundary) option which applies regardless of the type of scaling.

So my suggestion now would be:

* `--max-scale` and `--min-scale` (as in this PR)
* `--concurrency-target` (renaming the option)
* `--concurrency-limit` (adding to this PR, by setting the proper field in the spec)

Reply:
@markusthoemmes Just saw that `containerConcurrency` is part of `PodAutoscaler` which is not part of the public API. So is it correct, that this datum can't be changed from a client like `kn` ? Or is there a corresponding field on `Service` which the get put into the `PodAutoscaler` resource when a service is created ?

For now, I think for now we have to ommit `--concurrency-limit`